examples/pytorch/ogb/directional_GSN/preprocessing.py

```diff
@@ -0,0 +1,285 @@
+        else:
+            edge_lists.append(list(graphs_nx.edges))
+
+    for edge_list in edge_lists:
+        subgraph, orbit_partition, orbit_membership, aut_count = induced_edge_automorphism_orbits(edge_list=edge_list)
+        subgraph_dicts.append({'subgraph':subgraph, 'orbit_partition': orbit_partition, 
+                               'orbit_membership': orbit_membership, 'aut_count': aut_count})
+        orbit_partition_sizes.append(len(orbit_partition))
+        
+    ### load and preprocess dataset
+    dataset = DglGraphPropPredDataset(name=name, root=path)
+    split_idx = dataset.get_idx_split()
+        
+    # computation of subgraph isomoprhisms & creation of data structure
+    graphs_dgl = list()
+    for i, datapoint in tqdm(enumerate(dataset)):
+        g, label = datapoint
+        g = _prepare(g, subgraph_dicts)
+        graphs_dgl.append((g, label))
+
+    return graphs_dgl, orbit_partition_sizes, split_idx
+        
+def _prepare(g, subgraph_dicts):
+
+    edge_index = torch.stack(g.edges())
+    
+    identifiers = None
+    for subgraph_dict in subgraph_dicts:
+        counts = subgraph_isomorphism_edge_counts(edge_index, subgraph_dict)
+        identifiers = counts if identifiers is None else torch.cat((identifiers, counts),1) 
+
+    g.edata['subgraph_counts'] = identifiers.long()
+    
+    return g
+
+class pcbaDGL(torch.utils.data.Dataset):
+    def __init__(self, data, split):
+        self.split = split
+        
+        self.data = [data[split_ind] for split_ind in self.split]
+        
+        self.graph_lists = []
+        self.graph_labels = []
+        for g in self.data:
+            if g[0].number_of_nodes() > 5:
+                self.graph_lists.append(g[0])
+                self.graph_labels.append(g[1])
+        self.n_samples = len(self.graph_lists)
+        
+        new_graph_lists = []
+        for g in self.graph_lists:
+            g = self.get_subgraphs(g)
+            g.edata['eig'] = g.edata['eig'].float()
+            new_graph_lists.append(g)
+        self.graph_lists = new_graph_lists
+
+    def get_subgraphs(self, g):
+        vector_field = g.edata['subgraph_counts']
+        g.edata['eig'] = vector_field
+        return g
+
+    def __len__(self):
+        """Return the number of graphs in the dataset."""
+        return self.n_samples
+
+    def __getitem__(self, idx):
+        """
+            Get the idx^th sample.
+            Parameters
+            ---------
+            idx : int
+                The sample index.
+            Returns
+            -------
+            (dgl.DGLGraph, int)
+                DGLGraph with node feature stored in `feat` field
+                And its label.
+        """
+        return self.graph_lists[idx], self.graph_labels[idx]
+
+class pcbaDataset(Dataset):
+    def __init__(self,
+                 name,
+                 path='dataset/ogbg-molpcba',
+                 verbose=True,
```

Comment:
It seems that `verbose` is always True. Then it does not make sense to have this flag and related control flows.